src/linearcombination.jl

```diff
@@ -1,136 +1,86 @@
-    T = promote_type(eltype(A1), eltype(A2))
-    return LinearCombination{T}(tuple(A1, A2.maps...), tuple(one(T), map(-, A2.coeffs)...))
-end
-function Base.:(-)(A1::LinearCombination, A2::LinearMap)
-    size(A1) == size(A2) || throw(DimensionMismatch("-"))
-    T = promote_type(eltype(A1), eltype(A2))
-    return LinearCombination{T}(tuple(A1.maps..., A2), tuple(A1.coeffs..., -one(T)))
-end
-function Base.:(-)(A1::LinearMap, A2::LinearMap)
-    size(A1) == size(A2) || throw(DimensionMismatch("-"))
-    T = promote_type(eltype(A1), eltype(A2))
-    return LinearCombination{T}(tuple(A1, A2), tuple(one(T), -one(T)))
-end
-
-# scalar multiplication
-Base.:(-)(A::LinearMap) = LinearCombination{eltype(A)}(tuple(A), tuple(-one(eltype(A))))
-Base.:(-)(A::LinearCombination) = LinearCombination{eltype(A)}(A.maps, map(-, A.coeffs))
-
-function Base.:(*)(α::Number, A::LinearMap)
-    T = promote_type(eltype(α), eltype(A))
-    return LinearCombination{T}(tuple(A), tuple(α))
-end
-Base.:(*)(A::LinearMap, α::Number) = *(α, A)
-function Base.:(*)(α::Number, A::LinearCombination)
-    T = promote_type(eltype(α), eltype(A))
-    return LinearCombination{T}(A.maps, map(x->α*x, A.coeffs))
-end
-Base.:(*)(A::LinearCombination, α::Number) = *(α, A)
-
-function Base.:(\)(α::Number, A::LinearMap)
-    T = promote_type(eltype(α), eltype(A))
-    return LinearCombination{T}(tuple(A), tuple(1/α))
-end
-Base.:(/)(A::LinearMap, α::Number) = \(α, A)
-function Base.:(\)(α::Number, A::LinearCombination)
-    T = promote_type(eltype(α), eltype(A))
-    return LinearCombination{T}(A.maps, map(x->α\x, A.coeffs))
-end
-Base.:(/)(A::LinearCombination, α::Number) = \(α, A)
+Base.:(-)(A1::LinearMap, A2::LinearMap) = +(A1, -A2)
 
-# comparison of LinearCombination objects
-Base.:(==)(A::LinearCombination, B::LinearCombination) = (eltype(A)==eltype(B) && A.maps==B.maps && A.coeffs==B.coeffs)
+# comparison of LinearCombination objects, sufficient but not necessary
+Base.:(==)(A::LinearCombination, B::LinearCombination) = (eltype(A) == eltype(B) && A.maps == B.maps)
 
 # special transposition behavior
-LinearAlgebra.transpose(A::LinearCombination) = LinearCombination{eltype(A)}(map(transpose, A.maps), A.coeffs)
-LinearAlgebra.adjoint(A::LinearCombination)   = LinearCombination{eltype(A)}(map(adjoint, A.maps), map(conj, A.coeffs))
+LinearAlgebra.transpose(A::LinearCombination) = LinearCombination{eltype(A)}(map(transpose, A.maps))
+LinearAlgebra.adjoint(A::LinearCombination)   = LinearCombination{eltype(A)}(map(adjoint, A.maps))
 
 # multiplication with vectors
 function A_mul_B!(y::AbstractVector, A::LinearCombination, x::AbstractVector)
     # no size checking, will be done by individual maps
     A_mul_B!(y, A.maps[1], x)
-    A.coeffs[1] == 1 || lmul!(A.coeffs[1], y)
     l = length(A.maps)
     if l>1
         z = similar(y)
         for n=2:l
             A_mul_B!(z, A.maps[n], x)
-            axpy!(A.coeffs[n], z, y)
+            y .+= z
```

Comment:
Technically, we have our own 5-arg mul!, which would seem to do it in one pass. But for general `LinearMap`s, this would allocate a `z` in each call internally, and for matrices this is only in the making. I assume that allocating everytime is more expensive than visiting the preallocated components twice. But I agree, we should keep the idea of having an efficient "muladd!" in mind.